### DISCUSSION TITLE
How to use container on ghcr.io for action?

### DISCUSSION THREAD

**Discussion by lingfeiwang** [community/community#32065] [Actions]
Hi.

I'm having difficulty in autotesting the container from github action with a different action.

I could build and push the container and `docker pull` to a local machine. However, I tried to autotest the container with another action according to https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container#example-defining-credentials-for-a-container-registry, but could not make it work. I get the error below (with masked user and repo names)
```
/usr/bin/docker --config /home/runner/work/_temp/.docker_b0f7bff8-d134-4342-992c-5dbcb8f0abc7 login ghcr.io -u lingfeiwang --password-stdin
/usr/bin/docker --config /home/runner/work/_temp/.docker_b0f7bff8-d134-4342-992c-5dbcb8f0abc7 pull ghcr.io/user/repo
Error response from daemon: denied
Warning: Docker pull failed with exit code 1, back off 1.564 seconds before retry.
/usr/bin/docker --config /home/runner/work/_temp/.docker_b0f7bff8-d134-4342-992c-5dbcb8f0abc7 pull ghcr.io/user/repo
Error response from daemon: denied
Warning: Docker pull failed with exit code 1, back off 7.29 seconds before retry.
/usr/bin/docker --config /home/runner/work/_temp/.docker_b0f7bff8-d134-4342-992c-5dbcb8f0abc7 pull ghcr.io/user/repo
Error response from daemon: denied
Error: Docker pull failed with exit code 1
```

This seems an authentication issue. The container still remains private access at the testing phase. However, the same repo's action could push to the registry so read access should be included.

Have you met similar issues and how do you use a private container on ghcr.io for github action?

**Comment by mherger** [comment]
I'm using the `docker/login-action` to sign in before running my container:

```yaml
      - name: Login to GitHub Container Registry
        uses: docker/login-action@v2
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}

      - name: Run stuff
        run: docker run --rm ghcr.io/myorg/myimage:latest
```
